src/hello_client.erl

```diff
@@ -165,12 +166,17 @@ handle_call(terminate, _From, State) ->
 %% @hidden
 handle_info({?INCOMING_MSG, Message}, State) ->
     incoming_message(Message, State);
+handle_info(?PING, State = #client_state{ping_counter = PingCounter}) when PingCounter >= 3 ->
+    ?LOG_ERROR("There is no PONG answer on PING for a long time. Client will be restarted.", []),
```

Comment:
i do not see any added value by doing 3x ping instead of doing one. a reconnection should be triggered when a single ping/pong gets a delay greater then the threshold.